compiler/include/arg-helpers.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022-2023 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * This file contains various functions that are used by the command line
+ * argument parser. This parser is used by both Chapel and chapeldoc.
+ * Some of the functions in this header are duplicated in both and
+ * some have slightly different behavior.
+ **/
+
+#ifndef _ARG_HELPERS_H_
+#define _ARG_HELPERS_H_
+
+#include <cinttypes>
+#include <sys/stat.h>
+#include <cstdlib>
+#include <cstdio>
+#include <string>
+
+// The following is implemented in Chapel but we duplicate the code
+// in chpldoc; though ideally the common implementation would be shared
+// rather than duplicated. Due to default arguments we can't just redundantly
+// declare these so we control whether they're declared or not based on the
+// presence of a OUTSIDE_OF_CHPL_COMPILER macro
```

Comment:
> OUTSIDE_OF_CHPL_COMPILER

where is this defined? I can't find mention of it anywhere else but this comment

Reply:
Good catch, this comment mentions something I was thinking of doing but I ended up taking a different approach. So, I'll delete that last sentence. 

As an FYI:

The declaration for some of these functions (like `hexStr2uint64`) originally took default arguments. But apparently if you have multiple declarations of a function with default arguments (even of those defaults are the same) the compiler complains.
I was thinking maybe I could use a macro to avoid redundantly declaring those when compiling for Chapel, but I got lazy when it came time to figure out where I should `#define` it (and it seems like kind of a hack anyway). So instead I just took the defaults out and made all the uses of these functions in `arg.cpp` pass all their parameters.